tests/unit/s2n_init_test.c

```diff
@@ -77,6 +87,21 @@ int main(int argc, char **argv)
     EXPECT_EQUAL(pthread_join(init_thread, NULL), 0);
     EXPECT_SUCCESS(s2n_cleanup());
 
+    /* Calling s2n_init/s2n_cleanup in a different thread than s2n_cleanup_thread is called cleans up properly */
+    EXPECT_SUCCESS(s2n_init());
+    EXPECT_TRUE(s2n_is_initialized());
+    pthread_t s2n_cleanup_th = { 0 };
+    EXPECT_EQUAL(pthread_create(&s2n_cleanup_th, NULL, s2n_cleanup_thread_cb, NULL), 0);
+    EXPECT_EQUAL(pthread_join(s2n_cleanup_th, NULL), 0);
+    EXPECT_TRUE(s2n_is_initialized());
+    /* Calling s2n_cleanup_thread in the main thread leaves s2n initialized. */
+    EXPECT_SUCCESS(s2n_cleanup_thread());
+    EXPECT_TRUE(s2n_is_initialized());
+    EXPECT_SUCCESS(s2n_cleanup());
+    EXPECT_FALSE(s2n_is_initialized());
+    /* Second call to s2n_cleanup will fail, since the full cleanup is not idempotent. */
+    EXPECT_FAILURE_WITH_ERRNO(s2n_cleanup(), S2N_ERR_NOT_INITIALIZED);
```

Comment:
Oh, one last nit: our new test style is to use contexts to define individual tests for clarity. It makes it clearer where tests begin and end, and helps prevent them from accidentally affecting each other. Although idk if it can help prevent influence in this case :P

```suggestion
    /* Calling s2n_init/s2n_cleanup in a different thread than s2n_cleanup_thread is called cleans up properly */
    {
        EXPECT_SUCCESS(s2n_init());
        EXPECT_TRUE(s2n_is_initialized());
        pthread_t s2n_cleanup_th = { 0 };
        EXPECT_EQUAL(pthread_create(&s2n_cleanup_th, NULL, s2n_cleanup_thread_cb, NULL), 0);
        EXPECT_EQUAL(pthread_join(s2n_cleanup_th, NULL), 0);
        EXPECT_TRUE(s2n_is_initialized());
        /* Calling s2n_cleanup_thread in the main thread leaves s2n initialized. */
        EXPECT_SUCCESS(s2n_cleanup_thread());
        EXPECT_TRUE(s2n_is_initialized());
        EXPECT_SUCCESS(s2n_cleanup());
        EXPECT_FALSE(s2n_is_initialized());
        /* Second call to s2n_cleanup will fail, since the full cleanup is not idempotent. */
        EXPECT_FAILURE_WITH_ERRNO(s2n_cleanup(), S2N_ERR_NOT_INITIALIZED);
    }
```